src/language-js/comments.js

```diff
@@ -929,6 +929,56 @@ function getCommentChildNodes(node, options) {
   ) {
     return [...(node.decorators || []), node.key, node.value.body];
   }
+
+  // If `text` is undefined, it means that `getSortedChildNodes` was called for formatting a range,
+  // not for comment attachment.
+  if (node.type === "ForStatement" && text) {
+    const result = [];
+    for (const prop of ["init", "test", "update"]) {
+      result.push(
+        node[prop] ||
+          createCommentAnchorPseudoNode(
+            node,
+            prop,
+            result.length > 0
+              ? getNextNonSpaceNonCommentCharacterIndexWithStartIndex(
+                  text,
+                  nextIndexOf(";", text, locEnd(getLast(result))) + 1
+                )
+              : nextIndexOf(";", text, locStart(node))
+          )
+      );
+    }
+    result.push(node.body);
+    return result;
+  }
+}
+
+function nextIndexOf(char, text, startIdx) {
+  let nextIdx = startIdx;
+  for (;;) {
+    nextIdx = getNextNonSpaceNonCommentCharacterIndexWithStartIndex(
+      text,
+      nextIdx
+    );
+    if (nextIdx === false) {
+      /* istanbul ignore next */
+      return startIdx;
+    }
+    if (text.charAt(nextIdx) === char) {
+      return nextIdx;
+    }
+    nextIdx++;
+  }
+}
+
+function createCommentAnchorPseudoNode(enclosingNode, name, loc) {
+  return {
+    type: "Prettier:CommentAnchor",
```

Comment:
What do you think  `::CommentAnchor`, like a  css pseudo element. 😄

Reply:
I like it :+1: